src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1118,51 +1117,108 @@ def add_units_table(
-            )
-    aggregated_unit_properties = defaultdict()
-    for property_name in write_properties:
-        unit_col_args = dict(
-            name=property_name, description=property_descriptions.get(property_name, "No description.")
-        )
-        if property_name in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
-            unit_col_args.update(table=nwbfile.electrodes)
+    data_to_add = defaultdict(dict)
+    sorting_properties = checked_sorting.get_property_keys()
+    excluded_properties = list(skip_properties) + ["contact_vector"]
+    properties_to_extract = [property for property in sorting_properties if property not in excluded_properties]
 
-        units_table.add_column(**unit_col_args)
-        aggregated_unit_properties[property_name] = checked_sorting.get_property(key=property_name)
+    # Extract properties
+    for property in properties_to_extract:
+        data = checked_sorting.get_property(property)
+        index = isinstance(data[0], (list, np.ndarray, tuple))
+        description = property_descriptions.get(property, "No description.")
+        data_to_add[property].update(description=description, data=data, index=index)
+        if property in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
+            data_to_add[property].update(table=nwbfile.electrodes)
 
     # Unit name logic
     units_ids = checked_sorting.get_unit_ids()
-    if "unit_name" in aggregated_unit_properties:
-        unit_name_array = aggregated_unit_properties["unit_name"]
+    if "unit_name" in data_to_add:
+        unit_name_array = data_to_add["unit_name"]["data"]
     else:
         unit_name_array = units_ids.astype("str", copy=False)
-        unit_col_args = dict(
-            name="unit_name",
-            description="unique reference for each unit",
-        )
-        units_table.add_column(**unit_col_args)
-        aggregated_unit_properties["unit_name"] = unit_name_array
+        data_to_add["unit_name"].update(description="unique reference for each unit", data=unit_name_array, index=False)
 
-    for i, unit_id in enumerate(units_ids):
-        spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id, return_times=True)
-        kwargs = {key: val[i] for key, val in aggregated_unit_properties.items()}
+    # If the channel ids are integer keep the old behavior of asigning table's id equal to unit_ids
+    if np.issubdtype(units_ids.dtype, np.integer):
+        data_to_add["id"].update(data=units_ids.astype("int"), index=False)
 
-        # If the units ids are integer keep the old behavior of asigning table's id equal to unit_ids
-        if str(unit_id).isdigit():
-            id = int(unit_id)
-        else:
-            id = i
+    units_table_previous_properties = set(units_table.colnames) - {"spike_times"}
+    extracted_properties = set(data_to_add)
+    properties_to_add_by_rows = units_table_previous_properties | {"id"}
```

Comment:
Personally I'm not sure when it comes to readability. What's better, 

```python
set_c = set_a | set_b
```

or

```python
set_c = set_a.union(set_b)
```

?

Reply:
I never thought about this one. I am on the fence.
Does comparing a case with more elements case makes you more incline for one or the other?


`total_union = A | B | C | D`
vs
`total_union = A.union(B).union(C).union(D)`

Reply:
It's mostly that the pipe operator is so unique to sets and doesn't show up elsewhere, you have to mentally think about what it means if it's been a while since working with it.

I mean, but multiple unions you could also do it iteratively pairwise regardless (and probably should if it's dynamically sized or larger than 3 items)